remote/browse.go

```diff
@@ -99,7 +101,48 @@ func (rp LocalPreviews) Preview(ctx context.Context, _, refStr string) (*dataset
 		return nil, err
 	}
 
-	return base.CreatePreview(ctx, rp.fs, ref)
+	ds, err := dsfs.LoadDataset(ctx, rp.fs, ref.Path)
+	if err != nil {
+		log.Errorf("remote.Preview loading dataset: %w", err.Error())
+		return nil, err
+	}
+
+	ds.Name = ref.Name
+	ds.Peername = ref.Username
+
+	if ds.Transform != nil && ds.Transform.ScriptPath != "" {
+		f, err := rp.fs.Get(ctx, ds.Transform.ScriptPath)
+		if err != nil {
+			log.Errorf("remote.Preview loading transform script: %w", err.Error())
+			return nil, err
+		}
+		ds.Transform.SetScriptFile(f)
+	}
+
+	if ds.Readme != nil && ds.Readme.ScriptPath != "" {
+		f, err := rp.fs.Get(ctx, ds.Readme.ScriptPath)
+		if err != nil {
+			log.Errorf("remote.Preview loading readme script: %w", err.Error())
+			return nil, err
+		}
+		ds.Readme.SetScriptFile(f)
+	}
+
+	if ds.Viz != nil && ds.Viz.ScriptPath != "" {
+		f, err := rp.fs.Get(ctx, ds.Viz.ScriptPath)
+		if err != nil {
+			log.Errorf("remote.Preview loading viz script: %w", err.Error())
+			return nil, err
+		}
+		ds.Viz.SetScriptFile(f)
+	}
+
+	if err := ds.OpenBodyFile(ctx, rp.fs); err != nil {
+		log.Errorf("remote.Preview loading body file: %w", err.Error())
+		return nil, err
+	}
```

Comment:
am I crazy? Do we not already have a function that does this? aka loads the dataset & ensures that all the script files are loaded as well?

Reply:
yar.  `base.OpenDataset`

Reply:
ahhh ty, I was looking in `dsfs`